pkg/controllers/streaming/processor_controller.go

```diff
@@ -18,36 +18,225 @@ package streaming
 
 import (
 	"context"
-
+	"fmt"
 	"github.com/go-logr/logr"
+	"github.com/projectriff/system/pkg/apis/build/v1alpha1"
+	"github.com/projectriff/system/pkg/apis/streaming/v1alpha1/resources"
+	"github.com/projectriff/system/pkg/apis/streaming/v1alpha1/resources/names"
+	"github.com/projectriff/system/pkg/controllers/kmp"
+	"github.com/projectriff/system/pkg/tracker"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
 
+	buildv1alpha1 "github.com/projectriff/system/pkg/apis/build/v1alpha1"
 	streamingv1alpha1 "github.com/projectriff/system/pkg/apis/streaming/v1alpha1"
 )
 
 // ProcessorReconciler reconciles a Processor object
 type ProcessorReconciler struct {
 	client.Client
-	Log    logr.Logger
-	Scheme *runtime.Scheme
+	Log     logr.Logger
+	Scheme  *runtime.Scheme
+	Tracker tracker.Tracker
 }
 
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=streaming.projectriff.io,resources=processors/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 
 func (r *ProcessorReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("processor", req.NamespacedName)
+	ctx := context.Background()
+	log := r.Log.WithValues("processor", req.NamespacedName)
 
-	// your logic here
+	var original streamingv1alpha1.Processor
+	if err := r.Client.Get(ctx, req.NamespacedName, &original); err != nil {
+		return ctrl.Result{}, ignoreNotFound(err)
+	}
 
-	return ctrl.Result{}, nil
+	// Don't modify the informers copy
+	processor := original.DeepCopy()
+
+	// Reconcile this copy of the processor and then write back any status
+	// updates regardless of whether the reconciliation errored out.
+	result, err := r.reconcile(ctx, log, processor)
+
+	// check if status has changed before updating, unless requeued
+	if !result.Requeue && !equality.Semantic.DeepEqual(original.Status, processor.Status) {
+		if updateErr := r.Status().Update(ctx, processor); updateErr != nil {
+			log.Error(updateErr, "unable to update Processor status", "processor", processor)
+			return ctrl.Result{Requeue: true}, updateErr
+		}
+	}
+	return result, err
 }
 
 func (r *ProcessorReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	enqueueTrackedResources := &handler.EnqueueRequestsFromMapFunc{
```

Comment:
for consistency with other reconcilers, move `SetupWithManager` to the bottom of the file